packages/cli/src/databases/migrations/common/1723796243146-RefactorExecutionIndices.ts

```diff
@@ -31,12 +30,40 @@ import type { MigrationContext, ReversibleMigration } from '@/databases/types';
  *
  * - `deletedAt` for query at `ExecutionRepository.hardDeleteSoftDeletedExecutions`
  */
-export class RefactorExecutionIndices1717498465932 implements ReversibleMigration {
-	async up({ schemaBuilder, isPostgres, isSqlite, isMysql }: MigrationContext) {
-		await schemaBuilder.createIndex('execution_entity', ['status', 'startedAt']);
-		await schemaBuilder.createIndex('execution_entity', ['workflowId', 'status', 'startedAt']);
-		await schemaBuilder.createIndex('execution_entity', ['waitTill', 'status']);
-		await schemaBuilder.createIndex('execution_entity', ['stoppedAt', 'deletedAt', 'status']);
+export class RefactorExecutionIndices1723796243146 implements ReversibleMigration {
+	async up({ schemaBuilder, isPostgres, isSqlite, isMysql, runQuery, escape }: MigrationContext) {
+		if (isSqlite || isPostgres) {
+			const executionEntity = escape.tableName('execution_entity');
+
+			const workflowId = escape.columnName('workflowId');
+			const startedAt = escape.columnName('startedAt');
+			const waitTill = escape.columnName('waitTill');
+			const status = escape.columnName('status');
+			const deletedAt = escape.columnName('deletedAt');
+			const stoppedAt = escape.columnName('stoppedAt');
+
+			await runQuery(`
+				CREATE INDEX idx_execution_entity_workflow_id_started_at
+				ON ${executionEntity} (${workflowId}, ${startedAt})
+				WHERE ${startedAt} IS NOT NULL AND ${deletedAt} IS NULL;
+			`);
+
+			await runQuery(`
+				CREATE INDEX idx_execution_entity_wait_till_status_deleted_at
+				ON ${executionEntity} (${waitTill}, ${status}, ${deletedAt})
+				WHERE ${waitTill} IS NOT NULL AND ${deletedAt} IS NULL;
+			`);
+
+			await runQuery(`
+				CREATE INDEX idx_execution_entity_stopped_at_status_deleted_at
+				ON ${executionEntity} (${stoppedAt}, ${status}, ${deletedAt})
+				WHERE ${stoppedAt} IS NOT NULL AND ${deletedAt} IS NULL;
+			`);
+		} else if (isMysql) {
+			await schemaBuilder.createIndex('execution_entity', ['workflowId', 'startedAt']);
+			await schemaBuilder.createIndex('execution_entity', ['waitTill', 'status', 'deletedAt']);
+			await schemaBuilder.createIndex('execution_entity', ['stoppedAt', 'status', 'deletedAt']);
```

Comment:
Maybe these could be named explicitly as well, so the names would match for all different DBs

Reply:
Thank you, since we'll be dropping MySQL, I think this is okay for now.